psd-web/app/forms/why_reporting_form.rb

```diff
@@ -0,0 +1,78 @@
+class WhyReportingForm
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  attribute :hazard_type
+  attribute :hazard_description
+  attribute :non_compliant_reason
+  attribute :reported_reason_unsafe,             :boolean, default: false
+  attribute :reported_reason_non_compliant,      :boolean, default: false
+  attribute :reported_reason_safe_and_compliant, :boolean, default: false
+
+  validate :selected_at_least_one_checkbox
+  validate :mutually_exclusive_checkboxes
+  validates :non_compliant_reason, presence: true, if: -> { reported_reason_non_compliant }
+
+  def assign_to(investigation)
+    investigation.assign_attributes(
+      attributes
+        .slice("hazard_description", "hazard_type", "non_compliant_reason")
+        .merge(reported_reason: reported_reason, description: reason_created)
+    )
+  end
+
+private
+
+  def reported_reason
+    @reported_reason ||= case checkboxes
+                         when [true,  false, false] then Investigation.reported_reasons[:unsafe]
+                         when [false, true,  false] then Investigation.reported_reasons[:non_compliant]
+                         when [true,  true,  false] then Investigation.reported_reasons[:unsafe_and_non_compliant]
+                         when [false, false, true]  then Investigation.reported_reasons[:safe_and_compliant]
+                         end
```

Comment:
Please change: this is really unnecessary complication and can be much more simply expressed something like:

```suggestion
return :unsafe_and_non_compliant if reported_reason_unsafe && reported_reason_non_compliant
return :unsafe if reported_reason_unsafe
return :non_compliant if reported_reason_non_compliant
return :safe_and_compliant if reported_reason_safe_and_compliant
```
There's other ways this could be expressed but I think using the `checkboxes` in this way is quite difficult to follow.